app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  validates :uid, uniqueness: true, presence: true
+
+  def self.from_omniauth(params)
+    user = find_or_create_by(uid: params[:uid])
+    user.name = params[:info][:name]
+    user.email = params[:info][:email]
+    user.avatar = params[:info][:image]
```

Comment:
If the user does not have a photo loaded this will be a default image of their initial.